src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs

```diff
@@ -0,0 +1,632 @@
+		{
+			foreach (var expectedWarningsPerFile in expectedWarnings)
+			{
+				var actualWarningsPerFile = actualWarnings.FirstOrDefault(actualWarning => actualWarning.File.CompareWarningsFilePaths(expectedWarningsPerFile.File));
+				Assert.NotNull(actualWarningsPerFile,
+					$"Expected warnings file path '{expectedWarningsPerFile.File}' was not found.");
+
+				foreach (var expectedWarningsPerCode in expectedWarningsPerFile.WarningsPerCode)
+				{
+					var actualWarningsPerCode = actualWarningsPerFile!.WarningsPerCode.FirstOrDefault(x => x.Code == expectedWarningsPerCode.Code);
+					Assert.NotNull(actualWarningsPerCode,
+						$"Expected warning code '{expectedWarningsPerCode.Code}' was not found for the expected warnings file path '{expectedWarningsPerFile.File}'");
+
+					foreach (var expectedWarningsMessage in expectedWarningsPerCode.Messages)
+					{
+						Assert.True(actualWarningsPerCode!.Messages.Remove(expectedWarningsMessage),
+							$"Expected warning message '{expectedWarningsMessage}' was not found for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'");
+					}
+
+					Assert.AreEqual(0, actualWarningsPerCode!.Messages.Count,
+						$"Unexpected warning messages detected for the expected warnings file path '{expectedWarningsPerFile.File}' and warning code '{expectedWarningsPerCode.Code}'! Unexpected warning messages are: {string.Join("\n\t\t", actualWarningsPerCode.Messages)}");
+
+					actualWarningsPerFile.WarningsPerCode.Remove(actualWarningsPerCode);
+				}
+
+				Assert.AreEqual(0, actualWarningsPerFile!.WarningsPerCode.Count,
+					$"Unexpected warning codes detected for the expected warnings file path '{expectedWarningsPerFile.File}'! Unexpected warning codes are: {string.Join("\n\t\t", actualWarningsPerFile.WarningsPerCode.Select(c => c.Code).ToList())}");
+
+				actualWarnings.Remove(actualWarningsPerFile!);
+			}
+
+			Assert.AreEqual(0, actualWarnings.Count,
+				$"Unexpected warning files detected! Unexpected warning file paths are: {string.Join("\n\t\t", actualWarnings.Select(f => f.File).ToList())}");
+		}
+
+		#region Expected warning messages
+
+		// IMPORTANT: Always store expected File information as a relative path to the repo ROOT
+		private static readonly List<WarningsPerFile> expectedNativeAOTWarnings = new()
+		{
+			new WarningsPerFile
+			{
+				File = "ILC",
+				WarningsPerCode = new List<WarningsPerCode>
+				{
+					new WarningsPerCode
+					{
+						Code = "IL3050",
+						Messages = new List<string>
+						{
+							"<Module>..cctor(): Using member 'System.Enum.GetValues(Type)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload or the GetValuesAsUnderlyingType method instead.",
+						}
+					},
+				}
+			},
```

Comment:
@rmarinho This warning got fixed with: https://github.com/xamarin/xamarin-macios/pull/19516
And should be removed from the expected list of warnings. 

Do you want me to remove/test it, or you want to give it a go (basically selected lines should be deleted)?